readthedocs/core/resolver.py

```diff
@@ -134,25 +130,87 @@ def resolve_path(
             custom_prefix=custom_prefix,
         )
 
-    def resolve_domain(self, project, use_canonical_domain=True):
+    def resolve_version(self, project, version=None, filename="/"):
+        """
+        Get the URL for a specific version of a project.
+
+        If no version is given, the default version is used.
+
+        Use this instead of ``resolve`` if you have the version object already.
+        """
+        if not version:
+            default_version_slug = project.get_default_version()
+            version = project.versions(manager=INTERNAL).get(slug=default_version_slug)
```

Comment:
Do we want to default something here, instead of requiring a version? Sounds like this is mostly used when we already have a version?

Reply:
Do you mean making the version required? I like the behavior that we have in `resolve()`, where it also queries the default version if it isn't given. I do want to find usages of resolve where a version isn't given and use resolve_version instead, it would save us one query :D